docs/bokeh/source/docs/user_guide/interaction/widgets.rst

```diff
@@ -361,7 +376,7 @@ Select
 
 A single selection widget:
 
-.. bokeh-plot:: __REPO__/examples/interaction/widgets/select.py
+.. bokeh-plot:: __REPO__/examples/interaction/widgets/select_widget.py
```

Comment:
This inconsistency is unfortunate but there is a stdlib `select.py`, so having this name makes all the examples in the directory more or less un-runable. I looked at adding `_widget` to all of them but it's incredibly ugly and will make for some too-long filenames for gallery thumbnail cards.

Reply:
> (...)  having this name makes all the examples in the directory more or less un-runable.

Why is that and can this be fixed? This seems very fragile if we have to watch for every standard module name.

Reply:
```
dev-3.9 ❯ python slider.py
Traceback (most recent call last):
  File "/Users/bryan/work/bokeh/examples/interaction/widgets/slider.py", line 1, in <module>
    from bokeh.io import show
  File "/Users/bryan/work/bokeh/src/bokeh/__init__.py", line 56, in <module>
    __version__ = importlib.metadata.version("bokeh")
  File "/Users/bryan/anaconda/envs/dev-3.9/lib/python3.9/importlib/metadata.py", line 551, in version
    return distribution(distribution_name).version
  File "/Users/bryan/anaconda/envs/dev-3.9/lib/python3.9/importlib/metadata.py", line 257, in version
    return self.metadata['Version']
  File "/Users/bryan/anaconda/envs/dev-3.9/lib/python3.9/importlib/metadata.py", line 252, in metadata
    return email.message_from_string(text)
  File "/Users/bryan/anaconda/envs/dev-3.9/lib/python3.9/email/__init__.py", line 37, in message_from_string
    from email.parser import Parser
  File "/Users/bryan/anaconda/envs/dev-3.9/lib/python3.9/email/parser.py", line 12, in <module>
    from email.feedparser import FeedParser, BytesFeedParser
  File "/Users/bryan/anaconda/envs/dev-3.9/lib/python3.9/email/feedparser.py", line 27, in <module>
    from email._policybase import compat32
  File "/Users/bryan/anaconda/envs/dev-3.9/lib/python3.9/email/_policybase.py", line 9, in <module>
    from email.utils import _has_surrogates
  File "/Users/bryan/anaconda/envs/dev-3.9/lib/python3.9/email/utils.py", line 29, in <module>
    import socket
  File "/Users/bryan/anaconda/envs/dev-3.9/lib/python3.9/socket.py", line 54, in <module>
    import os, sys, io, selectors
  File "/Users/bryan/anaconda/envs/dev-3.9/lib/python3.9/selectors.py", line 12, in <module>
    import select
  File "/Users/bryan/work/bokeh/examples/interaction/widgets/select.py", line 1, in <module>
    from bokeh.io import show
  File "/Users/bryan/work/bokeh/src/bokeh/io/__init__.py", line 24, in <module>
    from .doc import curdoc
  File "/Users/bryan/work/bokeh/src/bokeh/io/doc.py", line 29, in <module>
    from ..document import Document
  File "/Users/bryan/work/bokeh/src/bokeh/document/__init__.py", line 50, in <module>
    from .document import DEFAULT_TITLE ; DEFAULT_TITLE
  File "/Users/bryan/work/bokeh/src/bokeh/document/document.py", line 51, in <module>
    from ..core.enums import HoldPolicyType
  File "/Users/bryan/work/bokeh/src/bokeh/core/enums.py", line 84, in <module>
    from .. import colors, palettes
  File "/Users/bryan/work/bokeh/src/bokeh/colors/__init__.py", line 25, in <module>
    from . import groups, named
  File "/Users/bryan/work/bokeh/src/bokeh/colors/groups.py", line 24, in <module>
    from .util import ColorGroup
  File "/Users/bryan/work/bokeh/src/bokeh/colors/util.py", line 27, in <module>
    from .color import RGB
  File "/Users/bryan/work/bokeh/src/bokeh/colors/color.py", line 39, in <module>
    from ..core.serialization import AnyRep, Serializable, Serializer
  File "/Users/bryan/work/bokeh/src/bokeh/core/serialization.py", line 45, in <module>
    import numpy as np
  File "/Users/bryan/anaconda/envs/dev-3.9/lib/python3.9/site-packages/numpy/__init__.py", line 144, in <module>
    from . import core
  File "/Users/bryan/anaconda/envs/dev-3.9/lib/python3.9/site-packages/numpy/core/__init__.py", line 100, in <module>
    from . import _add_newdocs_scalars
  File "/Users/bryan/anaconda/envs/dev-3.9/lib/python3.9/site-packages/numpy/core/_add_newdocs_scalars.py", line 9, in <module>
    import platform
  File "/Users/bryan/anaconda/envs/dev-3.9/lib/python3.9/platform.py", line 119, in <module>
    import subprocess
  File "/Users/bryan/anaconda/envs/dev-3.9/lib/python3.9/subprocess.py", line 227, in <module>
    _PopenSelector = selectors.SelectSelector
AttributeError: partially initialized module 'selectors' has no attribute 'SelectSelector' (most likely due to a circular import)
```
> Why is that and can this be fixed? This seems very fragile if we have to watch for every standard module name.

This is just standard python-path behavior. The only reasonable solution is to avoid the names, we will see examples fail in CI (which reminds me I should be able to add `examples/interaction` back now, I was not sure why it wa failing the other day, until now). 

The reason this is a problem for us in particular, and not really an issue for most python users, is that we have directories full of examples that are each individually meant to be run. there's no issue with having a `select.py` inside a package, e.g.